Microsoft.Toolkit.Uwp.UI.Animations/Behaviors/ViewportBehavior.cs

```diff
@@ -0,0 +1,203 @@
+            if (value)
+            {
+                obj.EnteredViewport?.Invoke(obj.AssociatedObject, EventArgs.Empty);
+            }
+            else
+            {
+                obj.ExitingViewport?.Invoke(obj.AssociatedObject, EventArgs.Empty);
+            }
+        }
+
+        private static void OnIsInViewportChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var obj = (ViewportBehavior)d;
+            var value = (bool)e.NewValue;
+            if (value)
+            {
+                obj.EnteringViewport?.Invoke(obj.AssociatedObject, EventArgs.Empty);
+            }
+            else
+            {
+                obj.ExitedViewport?.Invoke(obj.AssociatedObject, EventArgs.Empty);
+            }
+        }
+
+        private void ParentScrollViewer_ViewChanged(object sender, ScrollViewerViewChangedEventArgs e)
+        {
+            var associatedElementRect = AssociatedObject.TransformToVisual(_hostScrollViewer)
+                .TransformBounds(new Rect(0, 0, AssociatedObject.ActualWidth, AssociatedObject.ActualHeight));
+            var hostScrollViewerRect = new Rect(0, 0, _hostScrollViewer.ActualWidth, _hostScrollViewer.ActualHeight);
+
+            if (hostScrollViewerRect.Contains(new Point(associatedElementRect.Left, associatedElementRect.Top)) ||
+                hostScrollViewerRect.Contains(new Point(associatedElementRect.Right, associatedElementRect.Top)) ||
+                hostScrollViewerRect.Contains(new Point(associatedElementRect.Right, associatedElementRect.Bottom)) ||
+                hostScrollViewerRect.Contains(new Point(associatedElementRect.Left, associatedElementRect.Bottom)))
+            {
+                IsInViewport = true;
+
+                if (hostScrollViewerRect.Contains(new Point(associatedElementRect.Left, associatedElementRect.Top)) &&
+                    hostScrollViewerRect.Contains(new Point(associatedElementRect.Right, associatedElementRect.Top)) &&
+                    hostScrollViewerRect.Contains(new Point(associatedElementRect.Right, associatedElementRect.Bottom)) &&
+                    hostScrollViewerRect.Contains(new Point(associatedElementRect.Left, associatedElementRect.Bottom)))
+                {
+                    IsFullyInViewport = true;
+                }
+                else
+                {
+                    IsFullyInViewport = false;
+                }
+            }
+            else
+            {
+                IsInViewport = false;
+            }
+        }
+
+        private void Init()
+        {
+            _hostScrollViewer = GetScrollViewer();
+            if (_hostScrollViewer == null)
+            {
+                throw new InvalidOperationException("This behavior can only attach the element which is under a ScrollViewer.");
+            }
+
+            _hostScrollViewer.ViewChanged += ParentScrollViewer_ViewChanged;
+        }
+
+        private ScrollViewer GetScrollViewer()
```

Comment:
This method should be deleted. We already have helper methods like this in the toolkit.

Reply:
Done, use helper method instead.